front/src/edittracks/tracks.test.js

```diff
@@ -5,39 +5,147 @@ import {
+    await act(async () => {
+      render(
+        <Router>
+          <TrackCRUD
+            axios={axios}
+          />
+        </Router>,
+      );
+      await waitFor(() => expect(screen.getByText('Shooting Track 0')).toBeInTheDocument());
+      await waitFor(() => expect(screen.getByTitle('Add')).toBeInTheDocument());
+      fireEvent.click(screen.getByTitle('Add'));
+      await waitFor(() => expect(screen.getByPlaceholderText('Name')).toBeInTheDocument());
+      fireEvent.change(screen.getByPlaceholderText('Name'), {
+        target: {
+          value: 'new name',
+        },
+      });
+      fireEvent.change(screen.getByPlaceholderText('Description'), {
+        target: {
+          value: 'new description',
+        },
+      });
+      fireEvent.change(screen.getByPlaceholderText('Short description'), {
+        target: {
+          value: 'new short description',
+        },
+      });
+      await waitFor(() => expect(screen.getByPlaceholderText('Name').value).toBe('new name'));
+      await waitFor(() => expect(screen.getByTitle('Save')).toBeInTheDocument());
+      fireEvent.click(screen.getByTitle('Save'));
+      await waitFor(() => expect(axios.post).toHaveBeenCalled());
+      await waitFor(() => expect(screen.getByText('new name')).toBeInTheDocument());
+    });
+  });
+
+  it('should remove deleted track in list and call axios delete', async () => {
+    await act(async () => {
+      render(
+        <Router>
+          <TrackCRUD
+            axios={axios}
+          />
+        </Router>,
+      );
+      await waitFor(() => expect(screen.getByText('Shooting Track 0')).toBeInTheDocument());
+      fireEvent.click(screen.getByTitle('Delete'));
+      await waitFor(() => expect(screen.getByTitle('Save')).toBeInTheDocument());
+      fireEvent.click(screen.getByTitle('Save'));
+
+      await waitFor(() => expect(screen.queryByText('Shooting Track 0')).not.toBeInTheDocument());
+      await waitFor(() => expect(screen.queryByText('Rata poistettu')).toBeInTheDocument());
+      await waitFor(() => expect(axios.delete).toHaveBeenCalled());
+    });
+  });
+
+  it('should fail if no info given when adding track', async () => {
+    await act(async () => {
+      render(
+        <Router>
+          <TrackCRUD
+            axios={axios}
+          />
+        </Router>,
+      );
+      await waitFor(() => expect(screen.getByText('Shooting Track 0')).toBeInTheDocument());
+      await waitFor(() => expect(screen.getByTitle('Add')).toBeInTheDocument());
+      fireEvent.click(screen.getByTitle('Add'));
+      await waitFor(() => expect(screen.getByPlaceholderText('Name')).toBeInTheDocument());
+      await waitFor(() => expect(screen.getByTitle('Save')).toBeInTheDocument());
+      fireEvent.click(screen.getByTitle('Save'));
+      await waitFor(() => expect(axios.post).toHaveBeenCalled());
+      await waitFor(() => expect(screen.getByText('Radan lisäys epäonnistui')).toBeInTheDocument());
```

Comment:
Maybe a helper function would be handy in the future? This looks like hard work to manually write every single time

await normalWaitFor('Add', enum.Title)

or somesuch